Mathlib/GroupTheory/GroupAction/Defs.lean

```diff
@@ -733,6 +733,10 @@ theorem smul_zero (a : M) : a • (0 : A) = 0 :=
   SMulZeroClass.smul_zero _
 #align smul_zero smul_zero
 
+@[simp]
+lemma smul_boole (p : Prop) [Decidable p] (a : M) (b : A) :
+    (a • if p then b else 0) = if p then a • b else 0 := by rw [smul_ite, smul_zero]
```

Comment:
Shouldn't we have `b = 1` for this to be "boole"?

Reply:
Yeah sorry, I got confused because we can't actually state `smul_boole` (`a • 1` doesn't simplify to anything).